lib/actions/reassemble.js

```diff
@@ -2,7 +2,34 @@
 const { messages } = require('elasticio-node');
 const ObjectStorageWrapperExtended = require('./utils-wrapper/ObjectStorageWrapperExtended');
 
-async function processAction(msg) {
+let timeHandle;
+let groupList = [];
+let delay;
+
+async function timer(this_) {
+  for (let i = 0; i < groupList.length; i += 1) {
+    const storage = new ObjectStorageWrapperExtended(this_);
+    // eslint-disable-next-line no-await-in-loop
+    const results = await storage.lookupParsedObjectById(groupList[i]);
+    const incomingData = {};
+    results.messages.forEach((message) => {
+      incomingData[message.messageId] = message.messageData;
+    });
+
+    // eslint-disable-next-line no-await-in-loop
+    await storage.deleteObjectById(groupList[i]);
+
+    // eslint-disable-next-line no-await-in-loop
+    await this_.emit('data', messages.newMessageWithBody({
+      groupSize: results.messages.length,
+      groupId: results.messages[0].groupId,
+      messageData: incomingData,
+    }));
+  }
```

Comment:
Can `storage.lookupObjectsByQueryParameters([])` be used to fetch all the groups instead of manually trying to keep track of the group lists?

Reply:
To fetch all the groups using `lookupObjectsByQueryParameters([{ key: string, value: string }])` we need to define the `value` of each group and for that we will need to manually keep track on the different groups needed for that.
And `lookupObjectsByQueryParameters([])` has a current maximum of 5 items.
Let me know if am wrong!

Reply:
What is to stop us from setting the `value` for each group based on the flow ID and the step ID?

Reply:
After a discussion we've identified that this improvement can be done at a later point in time.